src/include/cache.i

```diff
@@ -352,6 +352,32 @@ __wt_eviction_updates_needed(WT_SESSION_IMPL *session, double *pct_fullp)
     return (bytes_updates > (uint64_t)(cache->eviction_updates_trigger * bytes_max) / 100);
 }
 
+/*
+ * __wt_btree_dominating_cache --
+ *     Return if a single btree is occupying at least half of any of our target's cache usage.
+ */
+static inline bool
+__wt_btree_dominating_cache(WT_SESSION_IMPL *session, WT_BTREE *btree)
+{
+    WT_CACHE *cache;
+    uint64_t bytes_max;
+
+    cache = S2C(session)->cache;
+    bytes_max = S2C(session)->cache_size + 1;
+
+    if (__wt_cache_bytes_plus_overhead(cache, btree->bytes_inmem) >
+      (uint64_t)(0.5 * cache->eviction_target * bytes_max) / 100)
+        return (true);
+    if (__wt_cache_bytes_plus_overhead(cache, btree->bytes_dirty_intl + btree->bytes_dirty_leaf) >
+      (uint64_t)(0.5 * cache->eviction_dirty_target * bytes_max) / 100)
+        return (true);
+    if (__wt_cache_bytes_plus_overhead(cache, btree->bytes_updates) >
+      (uint64_t)(0.5 * cache->eviction_updates_target * bytes_max) / 100)
```

Comment:
In this particular instance, it was the `eviction_updates_target` that caused the eviction server to kick in. But I think this rule should apply to any of the targets.